### PR TITLE
crc-32 and JSON ESM import fix

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/web3/web3.js/issues/6032
https://github.com/web3/web3.js/issues/6034

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-accounts/src/common/common.ts [hunk 1]
```diff
@@ -14,14 +14,14 @@ GNU Lesser General Public License for more details.
...
-import { buf as crc32Buffer } from 'crc-32';
+
 import { EventEmitter } from 'events';
 import type { Numbers } from 'web3-types';
 import { TypeOutput } from './types';
 import { intToBuffer, toType, parseGethGenesis } from './utils';
-import goerli from './chains/goerli.json';
-import mainnet from './chains/mainnet.json';
-import sepolia from './chains/sepolia.json';
+import goerli from './chains/goerli' ;
```
Comment (jdevcs) [inline packages/web3-eth-accounts/src/common/common.ts]: If we use `Static import with JSON import assertion` it solves problem for ESM only, but as it creates another issue for CJS build outputs due to current configuration settings that we have, so I converted json to ts., supporting both ESM+CJS hybrid builds in our case plus users dnt need additional `--experimental-json-modules`.

**Review by avkos — APPROVED** [review]

File: templates/.eslintignore.tmpl [hunk 2]
```diff
@@ -4,3 +4,6 @@ jest.config.js
 .eslintrc.js
 cypress
 cypress.config.js
+src/common/chains/**
+src/common/eips/**
+src/common/hardforks/**
```
Comment (jdevcs) [inline templates/.eslintignore.tmpl:9]: Ignoring all converted json files for eslint errors as we want no to minimum changes for now in these files ( for avoiding any breaking functionality and easy maintenance of these in near future by updating these json with released eth/comon json files)

**Comment by Spacesai1or** [comment]
Changes to `packages/web3-eth-accounts/src/common/common.ts` needs to account for the `Buffer` removal from #6033

**Review by luu-alex — APPROVED** [review]

**Comment by luu-alex** [comment]
LGTM, just need to fix conflicts and remove the [usage of buffer](https://github.com/web3/web3.js/pull/6049/files#diff-4e9a4fc0e6e2c7e2b4384459b2bba9f47edb57a8b7aa5f80bd2eac825b87ffc5R22). think this will be fix after fixing conflicts

**Comment by jdevcs** [comment]
some tests in geth http are failing, and will be fixed under https://github.com/web3/web3.js/issues/6060

**Pull request merged**